### ISSUE TITLE
Add/change length validation for preview summary text

### ISSUE THREAD

**Issue by mekline** [lookit/lookit-api#1075]
# TL;DR
Researchers are beginning to write summary text that is too long to fit nicely on the search page!  

# Narrative

As a family, I want to see a short, clear description of what I might learn about, not a wall of text that breaks up the search page.
As a reviewer (who will soon not be doing as much reviewer), I don't want to repeatedly tell researchers to make this text shorter.

# Acceptance Criteria

Accept when the new study/edit study form returns an error message telling the user to make the summary shorter if they submit a too-long text string!

# Implementation Notes

 The help text already tells researchers to keep this text under 100 words, but it would be okay if the validation gave them a bit of leeway, e.g. 150 words - the goal is brevity, not a tight limit.

**Comment by becky-gilbert** [comment]
@mekline not sure if you're aware, but apparently there's already a limit of 500 characters for the preview summary text:
https://github.com/lookit/lookit-api/blob/aa655a77126ec146b7121635120d50c2202e889d/studies/models.py#L298

I put together a 150-word limit and tried to test it with a 99-word string (which I was expecting to be valid), and I got a validation error due to the character count!

<img width="603" alt="Screenshot 2023-02-08 at 3 09 31 PM" src="https://user-images.githubusercontent.com/9041788/217673470-7dcfb2d0-a258-41f4-8c47-d0c1b5443b6f.png">
<img width="961" alt="Screenshot 2023-02-08 at 3 09 16 PM" src="https://user-images.githubusercontent.com/9041788/217673479-2f6e59a5-b838-4268-98c0-c1d838474441.png">

I tested this on production and got the same thing. So if you're seeing summaries that are too long, I suppose it's because this character limit is too high. 

So just to confirm, do you want me to add a word limit on top of the character limit (which might be confusing, because the word count could be valid while the character count isn't)? Or reduce the character limit?

p.s. here's a handy text generator for specific numbers of characters and words: https://www.blindtextgenerator.com/lorem-ipsum

**Comment by becky-gilbert** [comment]
And here's what it looks like when the word limit is reached:

<img width="700" alt="Screenshot 2023-02-08 at 3 43 05 PM" src="https://user-images.githubusercontent.com/9041788/217677002-efa879e6-2b64-4b89-a651-2b23d7ba8a97.png">

<img width="939" alt="Screenshot 2023-02-08 at 3 42 51 PM" src="https://user-images.githubusercontent.com/9041788/217677015-f822b699-8ad1-45b5-ba91-1971a89ed53d.png">

**Comment by mekline** [comment]
Empirically, most researchers are using statements that are between around 25 words/150 characters and 50 words/300 characters - so the occasional 100word one creates big whitespace gaps.  Let's reduce from 500 characters to 300 characters, and my preference would be to keep just the character count because as you said, the words-vs-characters limit is confusing to people IMO. (It's confusing to me at any rate.)

**Comment by becky-gilbert** [comment]
And here's what the form looks like with modified help text:
<img width="936" alt="Screenshot 2023-02-14 at 2 03 39 PM" src="https://user-images.githubusercontent.com/9041788/218873627-90d79e46-6c5c-4081-a863-2ba10502d3f0.png">

The text input field will now stop accepting characters after 300. Also, if someone's study already has a preview summary that is greater than 300 characters, then if they edit the study and try to save the form again, they will see the banner error message from [this comment above](https://github.com/lookit/lookit-api/issues/1075#issuecomment-1423376777) but with the 300 limit rather than 500.

Finally, I'm going to suggest that we don't actually change the character limit in the preview_summary column in the database, since I assume that would require a migration and would cause problems for existing studies that have preview summaries with 300+ characters.

**Comment by mekline** [comment]
Agree with all of this, I like it!

**Comment by becky-gilbert** [comment]
> Finally, I'm going to suggest that we don't actually change the character limit in the preview_summary column in the database, since I assume that would require a migration and would cause problems for existing studies that have preview summaries with 300+ characters.

@mekline so it turns out I was wrong about this. After discussing with @okaycj I think we do need to create a database migration. This means we need to handle the existing studies that have preview summaries with 300+ characters, otherwise the migration will fail.

Here are some options:

- Create a programatic solution as part of the migration, e.g. if any existing summaries are too long, then cut them off after a certain number of characters (or after the last period within the character limit). Of course we would need to let researchers know that we're doing this, and then follow up with the specific researchers whose studies were affected.

- Do it in a few stages: 
  1. Add the character limit text and warning to the form now (to prevent researchers from writing summaries that are too long in new studies).
  2. Reach out to researchers who have studies with summaries that are over limit and ask them to reduce the length.
  3. After all of the existing summaries are within the character limit, run the database migration.

Any thoughts or other suggestions?

**Comment by becky-gilbert** [comment]
After f2f discussion, we have decided on the following solution:
- Run a migration that edits the existing summaries that are over the character limit. Specifically:
    - the summary should be clipped at 297 characters
    - a "..." should be added to the end of the summary
    - the (clipped?) summary text should be appended to the start of the main study description
- Notify any researchers whose studies are affected, so that they can edit their study summary and/or description if desired.

Questions for @mekline: 
1. When a summary is clipped, are we appending the whole summary to the start of the main study description, or just the text that that was clipped off?
2. If a study is active and the researcher wants to edit their study summary in response to this change, then that will trigger a study rejection and require admin re-review, right? If so, perhaps we should flag this when we notify researchers and coordinate with them to quickly approve any studies that get rejected.

**Comment by mekline** [comment]
1. The whole text, because the preview text that appears on the search page doesn't appear anywhere on the study detail apge
2. Yes! Let's make sure we include this in the announcement and make sure Mark knows too!
